ciao-controller/internal/datastore/datastore.go

```diff
@@ -1239,7 +1183,15 @@ func (ds *Datastore) GetInstanceLastStats(nodeID string) types.CiaoServersStats
 		if instance.NodeID != nodeID {
 			continue
 		}
-		serversStats.Servers = append(serversStats.Servers, instance)
+
+		i, err := ds.GetInstance(instance.ID)
+		if err != nil {
+			continue
```

Comment:
It is highly unlikely this would ever happen. I don't know if there's much value in logging it, but we can certainly do that.